PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/SingleOperationInLoopCheck.java

```diff
@@ -0,0 +1,288 @@
+        public SingleOperationInLoopVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+            this.isOnTheFly = isOnTheFly;
+        }
+
+        /**
+         * Visit the for statement and check for single Azure client operation inside the loop.
+         * @param statement The for statement to check
+         */
+        @Override
+        public void visitForStatement(@NotNull PsiForStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+        /**
+         * Visit the foreach statement and check for single Azure client operation inside the loop.
+         * @param statement The foreach statement to check
+         */
+        @Override
+        public void visitForeachStatement(@NotNull PsiForeachStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+        /**
+         * Visit the while statement and check for single Azure client operation inside the loop.
+         * @param statement The while statement to check
+         */
+        @Override
+        public void visitWhileStatement(@NotNull PsiWhileStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+        /**
+         * Visit the do-while statement and check for single Azure client operation inside the loop.
+         * @param statement The do-while statement to check
+         */
+        @Override
+        public void visitDoWhileStatement(@NotNull PsiDoWhileStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+
+        /**
+         * Check the loop statement for a single Text Analytics Azure client operation inside the loop.
+         * @param loopStatement The loop statement to check
+         */
+        private boolean checkLoopForTextAnalyticsClientOperation(PsiStatement loopStatement) {
+
+            // extract body of the loop
+            PsiStatement loopBody = getLoopBody(loopStatement);
+
+            if (loopBody == null) {
+                return false;
+            }
+
+            // Extract the code block from the block statement
+            PsiBlockStatement blockStatement = (PsiBlockStatement) loopBody;
+            PsiCodeBlock codeBlock = blockStatement.getCodeBlock();
+
+            // extract statements in the loop body
+            for (PsiStatement statement : codeBlock.getStatements()) {
+
+                    // Check if the statement is an expression statement and is an Azure client operation
+                if (statement instanceof PsiExpressionStatement) {
+                        isExpressionAzureClientOperation(statement);
+                }
+
+                // Check if the statement is a declaration statement and is an Azure client operation
+                if (statement instanceof PsiDeclarationStatement) {
+                        isDeclarationAzureClientOperation((PsiDeclarationStatement) statement);
```

Comment:
Same here, we should not be directly casting and avoid potential ClassCastExceptions.

Reply:
I used instanceof so there won't be exceptions:)